Assets/Scripts/Cards/CardManager.cs

```diff
@@ -71,35 +59,52 @@ public void Init()
                 var randomIndex = Random.Range(0, cardDatas.Count);
                 (cardDatas[i], cardDatas[randomIndex]) = (cardDatas[randomIndex], cardDatas[i]);
             }
+
+            UpdateDeckCardCount();
+
+            barGroup = GameObject.Find("CardBarGroup").GetComponent<CardBarGroup>();
+            if (barGroup == null)
+                Debug.LogError("BarGroup이 씬 내 존재하지 않습니다.");
+            barGroup.Init(onSelectedCardsCountChanged: UpdateCanUseCard);
         }
 
         #region Using Card
 
-        private void UpdateCanUseCard()
+        private void UpdateDeckCardCount()
         {
-            canUseCard = TurnManager.Instance.gameState == GameState.PlayerInput
-                ? barGroup.selectedCards.Count > 0
-                : false;
-
-            useCardButton.interactable = canUseCard;
+            var args = new RemainingCardChangeArgs();
```

Comment:
IDispoable 구현해둬서 Dispose를 실행해 주시면 좋아요.
해당 이벤트객체를 저장하는 경우가 아니면 꼭 args.Dispose()를 실행해주세요. 풀에 반환합니다.
using 구문 이용하면 편하게 할 수 있어요
```suggestion
            using RemainingCardChangeArgs args = RemainingCardChangeArgs.Get();
```
{}로 스코프 지정할 수 있고, 지정안하면 함수 종료시 자동으로 Dispose가 실행됩니다

Reply:
아하 넵 확인했습니다 감사합니다~